lib/index.ts

```diff
@@ -57,17 +59,23 @@ class PWMetrics {
       if (this.expectations) {
         expectations.validateMetrics(this.expectations);
         this.expectations = expectations.normalizeMetrics(this.expectations);
-      } else throw new Error(getMessageWithPrefix('ERROR', 'NO_EXPECTATIONS_FOUND'));
+      } else throw new Error(messages.getMessageWithPrefix('ERROR', 'NO_EXPECTATIONS_FOUND'));
     }
   }
 
   async start() {
     const runs = Array.apply(null, {length: +this.runs}).map(Number.call, Number);
     let metricsResults: MetricsResults[] = [];
 
+    let resultHasExpectationErrors = false;
+
     for (let runIndex of runs) {
       try {
-        metricsResults[runIndex] = await this.run();
+        const currentMetricResult: MetricsResults = await this.run();
+        if (!resultHasExpectationErrors) {
+          resultHasExpectationErrors = this.resultHasExpectationErrors(currentMetricResult);
```

Comment:
It would be better if `this.resultHasExpectationErrors` threw an error when `this.flag.expectations` is on, don't u think? This way, expectation errors would be handled like other errors and this `this.run` execution would be flagged as `FAILED_RUN` instead of `SUCCESS_RUN`